sdk/python/foundation-models/system/inference/text-to-image/aacs-prepare-uai.ipynb

```diff
@@ -0,0 +1,448 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "role_name = \"Cognitive Services Contributor\"  # minimum role required for listing AACS access keys\n",
+    "\n",
+    "\n",
+    "def assign_access_to_acs(scope):\n",
+    "    role_defs = role_definition_client.role_definitions.list(scope=scope)\n",
+    "    role_def = next((r for r in role_defs if r.role_name == role_name))\n",
+    "\n",
+    "    from azure.core.exceptions import ResourceExistsError\n",
+    "\n",
+    "    try:\n",
+    "        role_assignment_client.role_assignments.create(\n",
+    "            scope=scope,\n",
+    "            role_assignment_name=str(uuid.uuid4()),\n",
+    "            parameters=RoleAssignmentCreateParameters(\n",
+    "                role_definition_id=role_def.id,\n",
+    "                principal_id=uai_principal_id,\n",
+    "                principal_type=\"ServicePrincipal\",\n",
+    "            ),\n",
+    "        )\n",
+    "    except ResourceExistsError as ex:\n",
+    "        pass\n",
+    "    except Exception as ex:\n",
+    "        print(ex)\n",
+    "        raise ex\n",
+    "\n",
+    "    if role_name in uai_role_check_list:\n",
+    "        del uai_role_check_list[role_name]\n",
+    "    print(\n",
+    "        f\"Role assignment for {role_name} at the Azure AI Content Safety resource level completed.\"\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 2.4.3 Grant the user identity access to the Azure Content Safety resource"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.mgmt.cognitiveservices import CognitiveServicesManagementClient\n",
+    "\n",
+    "acs_client = CognitiveServicesManagementClient(credential, subscription_id)\n",
+    "\n",
+    "\n",
+    "def find_acs(accounts):\n",
+    "    return next(x for x in accounts if x.kind == \"ContentSafety\")\n",
+    "\n",
+    "\n",
+    "if aacs_name == \"\":\n",
+    "    for acs_resource in acs_client.accounts.list_by_resource_group(resource_group):\n",
+    "        assign_access_to_acs(acs_resource.id)\n",
+    "else:\n",
+    "    acs_resource = acs_client.accounts.get(resource_group, aacs_name)\n",
+    "    assign_access_to_acs(acs_resource.id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 2.4.4 Assign AcrPull at the workspace container registry scope\n",
+    "Since we will create the Azure AI Content Safety enabled text-to-image endpoint with User Assigned Identity, the user managed identity must have Storage Blob Data Reader permission on the storage account for the workspace, and AcrPull permission on the Azure Container Registry (ACR) for the workspace. Make sure your User Assigned Identity has the right permission."
```

Comment:
```suggestion
    "Since we will create the Azure AI Content Safety (AACS) enabled text-to-image endpoint with User Assigned Identity, the user managed identity must have Storage Blob Data Reader permission on the storage account for the workspace, and AcrPull permission on the Azure Container Registry (ACR) for the workspace. Make sure your User Assigned Identity has the right permission."
```